apps/web/src/pages/integrations/UpdateProviderPage.tsx

```diff
@@ -103,229 +112,116 @@ export function UpdateProviderPage() {
     });
   }, [integrationId, providers]);
 
-  if (isLoading || areEnvironmentsLoading) {
-    return (
-      <SideBarWrapper>
-        <Stack
-          align="center"
-          justify="center"
-          sx={{
-            height: '100%',
-          }}
-        >
-          <Loader color={colors.error} size={32} />
-        </Stack>
-      </SideBarWrapper>
-    );
-  }
+  const onFrameworkClickCallback = (newFramework: FrameworkEnum) => {
+    setSidebarState('expanded');
+    setFramework(newFramework);
+  };
+
+  const onClose = () => {
+    navigate('/integrations');
+  };
 
-  if (selectedProvider === null) {
-    return null;
-  }
+  const onBack = () => {
+    if (sidebarState === 'expanded') {
+      setSidebarState('normal');
+      setFramework(null);
+    }
+  };
 
   return (
-    <SideBarWrapper>
-      <Form name={'connect-integration-form'} noValidate onSubmit={handleSubmit(onUpdateIntegration)}>
-        <Group spacing={5}>
-          <ProviderImage providerId={selectedProvider?.providerId} />
-          <Controller
-            control={control}
-            name="name"
-            defaultValue=""
-            render={({ field }) => {
-              return (
-                <NameInput
-                  {...field}
-                  value={field.value ? field.value : selectedProvider?.displayName}
-                  data-test-id="provider-instance-name"
-                  placeholder="Enter instance name"
-                />
-              );
-            }}
-          />
-          <ActionIcon
-            variant={'transparent'}
-            onClick={() => {
-              navigate('/integrations');
-            }}
-          >
-            <Close color={colors.B40} />
-          </ActionIcon>
-        </Group>
-        <Group mb={16} mt={16} spacing={16}>
-          <IntegrationChannel
-            name={CHANNEL_TYPE_TO_STRING[selectedProvider?.channel || ChannelTypeEnum.EMAIL]}
-            type={selectedProvider?.channel || ChannelTypeEnum.EMAIL}
-          />
-          <IntegrationEnvironmentPill
-            name={
-              environments?.find((environment) => environment._id === selectedProvider?.environmentId)?.name ||
-              'Development'
-            }
-          />
-        </Group>
-        <CenterDiv>
-          <When truthy={!haveAllCredentials}>
-            <WarningMessage spacing={12}>
-              <DisconnectGradient />
-              <div>
-                Set up credentials to start sending notifications
-                <br />
+    <FormProvider {...methods}>
+      <Sidebar
+        isOpened={!!selectedProvider}
+        isLoading={isLoading || areEnvironmentsLoading}
+        isExpanded={sidebarState === 'expanded'}
```

Comment:
I saw the `expanded` string to handle the state. Might worth to define it in a constant or enum if we have more states (I guess `normal` at least; just found it).